modules/dgi_actions_handle/modules/dgi_actions_handle_constraints/dgi_actions_handle_constraints.module

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/**
+ * @file
+ * Contains the dgi_actions_handle_constraints module.
+ */
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Entity\ContentEntityTypeInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Implements hook_entity_base_field_info_alter().
+ *
+ * Adds the UniqueField, Required constraint to the suffix and identifer fields.
+ */
+function _dgi_actions_handle_constraints_suffix_validation_add_constraint(array &$fields, EntityTypeInterface $entity_type): void {
+  try {
+    $config = \Drupal::config('dgi_actions_handle_constraints.settings');
+    $constraint_settings = $config->get('constraint_settings');
+    // Constraint settings is an array of values
+    // for each field that should be modified.
+    foreach ($constraint_settings as $constraintSetting) {
+      if (!isset($fields[$constraintSetting['field_name']]) || $constraintSetting['entity_type'] !== $entity_type->id()) {
+        continue;
+      }
+      // Both identifier and suffix fields should be unique.
+      $fields[$constraintSetting['field_name']]->addConstraint('UniqueField');
+
+      // If the field is a suffix field, set it as required.
+      if ($constraintSetting['field_usage'] === 'suffix') {
+        $fields[$constraintSetting['field_name']]->setRequired(TRUE);
```

Comment:
So, does this not run into the similar issues as discussed before around the contextual nature of how things get evaluated and the potential multiple scenarios to which an identifier can be minted?

That is take the example config with `field_pid`. There would be an expectation that all migrated items would have it and have it be set. For context,`field_pid` is intended to store the mapping from FCREPO during a site being migrated from I7. 

Post migration `field_pid` would no longer be intended to be used, but still would exist on the content type. Given the example configuration any new items that were born in the D10 repo would still be requiring this field and failing validation.

Reply:
This module isn't enabled by default and would only be enabled when a customer needs it with the config for whatever field is being used to mint Handles (see the capitalisation :P ) ? The expectation would be to enable it before Handle  minting has been done. If it's anything else, it's the customer's responsibility to ensure that the data is correct.

Reply:
So this makes the suffix field required always, wouldn't it just be better to do this in the content type then?

Reply:
Updated readme